### PR TITLE
Added the KillProcessWrapper script

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [48242](https://github.com/demisto/etc/issues/48242)

## Description
Added the KillProcessWrapper script which is a cross-vendor wrapper script that triggers a ‘process kill’ command - i.e executes the proper kill process command according to the vendor: CrowdstrikeFalcon or Cortex XDR.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [x] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    Args:
+        endpoint_id: The Endpoint ID in which you would like to kill the given process.
+        process_id: The ID of the process to kill.
+        process_name: The name of the process to kill.
+
+    Returns: A list of Commands.
+    """
+    commands = []
+
+    if process_name and not process_id:
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:34]: cant we just do two ifs here?

```
if processName:
    append xdr
if processID 
   append crowdstrike
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+    if approve_action == 'YES':
+        try:
+            # Handle missing arguments:
+            if not endpoint_id:
+                raise Exception('Endpoint ID was not specified.')
+            if not (process_id or process_name):
+                raise Exception('Process information was not specified - '
+                                'You should provide the process ID or the process name.')
+            return_results(run_killing_process_action(endpoint_id, process_id, process_name))
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:88]: Whats the case when we run both but only 1 installed, or 1 failed (as expected). This wont return as an error? I think we would only want to return an error if no successful commands ran

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+args:
+- default: false
+  description: ' The Endpoint ID in which you would like to kill the given process.'
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:3]:
```suggestion
  description: 'The Endpoint ID in which you would like to kill the given process.'
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  version: -1
+enabled: false
+name: KillProcessWrapper
+script: '-'
+subtype: python3
+system: false
+timeout: '0'
+type: python
+dockerimage: demisto/python3:3.10.4.29342
+fromversion: 6.2.0
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:44]: This cant be 5.0.0?

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+| --- | --- |
+| Script Type | python3 |
+| Cortex XSOAR Version | 6.2.0 |
+
+## Inputs
+---
+
+| **Argument Name** | **Description** |
+| --- | --- |
+| endpoint_id |  The Endpoint ID in which you would like to kill the given process. |
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/README.md:16]:
```suggestion
| endpoint_id | The Endpoint ID in which you would like to kill the given process. |
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+args:
+- default: false
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:2]: does this make sense as a default value?

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  secret: false
+- default: false
+  description: 'The name of the process to kill. '
+  isArray: false
+  name: process_name
+  required: false
+  secret: false
+- auto: PREDEFINED
+  default: false
+  description: 'Do you confirm killing the process? '
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:22]:
```suggestion
  description: 'Are you sure you want to kill the process? '
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 8]
```diff
@@ -0,0 +1,46 @@
+args:
+- default: false
+  description: ' The Endpoint ID in which you would like to kill the given process.'
+  isArray: false
+  name: endpoint_id
+  required: true
+  secret: false
+- default: false
+  description: 'The ID of the process to kill. '
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:9]: I think we should make it clear that either the id or the name must be provided.

**Review by JudahSchwartz — COMMENTED** [review]
Really nice tests

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py (hunk 1 above)
Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:34]: You are right - Changed that. Thanks!

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml (hunk 4 above)
Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:44]: As far as I know, new content should be created for versions 6.2.0 and up. 
Let me know if you know something else.

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml (hunk 8 above)
Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:9]: Added that.

File: Packs/MalwareInvestigationAndResponse/pack_metadata.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/pack_metadata.json:1]: Added that - thanks.
(I consulted with David as well).

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py (hunk 2 above)
Comment (JudahSchwartz) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:88]: We should also check, is there any message displayed after user clicks the button? (Action finished successfully, or action failed, popup)

Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:88]: The readable output is displayed in the incident war room... I think that is satisfying.

Comment (JudahSchwartz) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:88]: Thats how the other buttons currently work?

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+from typing import List
+
+XDR_PROCESS_KILL_COMMAND = 'xdr-run-script-kill-process'
+CROWDSTRIKE_PROCESS_KILL_COMMAND = 'cs-falcon-rtr-kill-process'
+
+''' STANDALONE FUNCTION '''
+
+
+def create_commands(endpoint_id: str = None,
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:18]:
```suggestion
def create_commands(endpoint_id: str,
```

Comment (JudahSchwartz) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:18]: since this must be provided

Comment (ShacharKidor) [reply Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.py:18]: Changed that.

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 11]
```diff
@@ -29,18 +31,43 @@ args:
...
 name: KillProcessWrapper
+outputs:
+- contextPath: CrowdStrike.Command.kill
+  description: The outputs of the CrowdStrike kill process command.
+  type: List
+- contextPath: CrowdStrike.Command.kill.Error
+  description: The status of the CrowdStrike kill process command.
+  type: String
+- contextPath: CrowdStrike.Command.kill.HostID
+  description: The endpint ID of the process.
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:48]:
```suggestion
  description: The endpoint ID of the process.
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 12]
```diff
@@ -29,18 +31,43 @@ args:
...
   id: KillProcessWrapper
   version: -1
 enabled: false
 name: KillProcessWrapper
+outputs:
+- contextPath: CrowdStrike.Command.kill
+  description: The outputs of the CrowdStrike kill process command.
+  type: List
+- contextPath: CrowdStrike.Command.kill.Error
+  description: The status of the CrowdStrike kill process command.
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:45]:
```suggestion
  description: The error status of the CrowdStrike kill process command.
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 13]
```diff
@@ -1,25 +1,27 @@
...
 - default: false
-  description: ' The Endpoint ID in which you would like to kill the given process.'
+  description: The Endpoint ID in which you would like to kill the given process.
   isArray: false
   name: endpoint_id
   required: true
   secret: false
 - default: false
-  description: 'The ID of the process to kill. '
+  description: The ID of the process to kill. The process ID or the process name must
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:9]:
```suggestion
  description: The ID of the process to kill. Either the process_id or the process_name must
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 14]
```diff
@@ -1,25 +1,27 @@
...
+    be specified.
   isArray: false
   name: process_id
   required: false
   secret: false
 - default: false
-  description: 'The name of the process to kill. '
+  description: The name of the process to kill. The process name or the process ID
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:16]:
```suggestion
  description: The name of the process to kill. Either the process_id or the process_name
```

File: Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml [hunk 15]
```diff
@@ -1,25 +1,27 @@
...
+    must be specified.
   isArray: false
   name: process_name
   required: false
   secret: false
 - auto: PREDEFINED
   default: false
-  description: 'Do you confirm killing the process? '
+  description: Are you sure you want to kill the process?
```
Comment (JudahSchwartz) [inline Packs/MalwareInvestigationAndResponse/Scripts/KillProcessWrapper/KillProcessWrapper.yml:24]:
```suggestion
  description: Are you sure you want to kill this process?
```

**Review by JudahSchwartz — APPROVED** [review]
code looks good, a couple of doc changes
Nice job on this one =D

**Pull request merged**